File: robocode-tankroyale-server/src/main/java/dev/robocode/tankroyale/server/events/BotHitWallEvent.java
package dev.robocode.tankroyale.server.events;

import lombok.Value;

/**
 * Event sent when a bot has hit a wall.
 * 
 * @author Flemming N. Larsen
 */
@Value
public class BotHitWallEvent implements Event {

	/** Turn number when event occurred */
	int turnNumber;

	/** Bot id of the victim that has hit a wall */
	int victimId;
}

File: robocode-tankroyale-server-kotlin/src/main/kotlin/dev/robocode/tankroyale/server/event/BotHitWallEvent.kt
package dev.robocode.tankroyale.server.event

/** Event sent when a bot has hit a wall. */
data class BotHitWallEvent(
    /** Turn number when event occurred */
    override val turnNumber: Int,

    /** Bot id of the victim that has hit a wall */
    val victimId: Int,

) : Event()